drainer/server.go

```diff
@@ -216,7 +216,7 @@ func (s *Server) heartbeat(ctx context.Context) <-chan error {
 	s.tg.Go("heartbeat", func() {
 		defer func() {
 			close(errc)
-			s.Close()
+			defer func() { go s.Close() }()
```

Comment:
Can be just `go s.Close()`?

Reply:
update by https://github.com/pingcap/tidb-binlog/pull/674/commits/ad0a4a08a308d0e1ee6e994d82cdc4a0249b1396